flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/StateHandleDownloadSpec.java

```diff
@@ -53,7 +53,9 @@ public IncrementalLocalKeyedStateHandle createLocalStateHandleForDownloadedState
         return new IncrementalLocalKeyedStateHandle(
                 stateHandle.getBackendIdentifier(),
                 stateHandle.getCheckpointId(),
-                new DirectoryStateHandle(downloadDestination),
+                // This handle is just temporary and will the directory size will never be used for
+                // metrics, so we just fill it in as 0.
+                new DirectoryStateHandle(downloadDestination, 0L),
```

Comment:
For completeness, why don't you pass here `stateHandle.getStateSize()`?

Reply:
Need to double check if that would do the right thing, I didn't care much because none of that is every reported, but yeah...if it works it's nicer.